# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/Lowering.cpp
# Commit: eb35fdc50799
# Full Hash: eb35fdc507996da61eaceffeb9f9e9b12f3ba0ca
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-10 15:21:10
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.2: Establish dependency from tls for x86 callWithABI div/mod i64. r=lth
#   
#   x86 has few register so to do div/mod for i64 it call the runtime
#   and clobber almost all gp registers including WasmTlsReg.
#   To be able to call c++ runtime via Builtin thunk we need to set up WasmTlsReg.
# ==============================================================================

diff -r 3952f1301e38 -r eb35fdc50799 js/src/jit/Lowering.cpp
--- a/js/src/jit/Lowering.cpp	Thu Sep 10 02:41:27 2020 +0000
+++ b/js/src/jit/Lowering.cpp	Thu Sep 10 05:18:04 2020 +0000
@@ -1845,6 +1845,14 @@
   MOZ_CRASH("Unhandled number specialization");
 }
 
+void LIRGenerator::visitWasmBuiltinDivI64(MWasmBuiltinDivI64* div) {
+  lowerWasmBuiltinDivI64(div);
+}
+
+void LIRGenerator::visitWasmBuiltinModI64(MWasmBuiltinModI64* mod) {
+  lowerWasmBuiltinModI64(mod);
+}
+
 void LIRGenerator::visitMod(MMod* ins) {
   MOZ_ASSERT(ins->lhs()->type() == ins->rhs()->type());
   MOZ_ASSERT(IsNumberType(ins->type()));